### ISSUE TITLE
Calendar is not working 

### ISSUE THREAD

**Issue by antbald** [patchthecode/JTAppleCalendar#1068]
**(Required) Version Number:**

## Description
The module is not working anymore. Is completely freezed. The example is not working either.

## Steps To Reproduce
Tap on the dates and it is not getting the input

**Comment by felipeflorencio** [comment]
The problem is how was declared the type and the protocol.

Now we have a default implementation in the protocol, just for convenience so not everyone implement.

But this create a problem, and the problem is, when we say that our delegate is a protocol type, when you have a default implementation, like he did in the extension, even if in our concrete class we implement that variable will actually call our protocol, as the variable it self is the type protocol.

The solution will need a little change to get back to work looks like

Nice article about why this is wrong and happening: https://medium.com/@xcadaverx/default-protocol-extension-caveats-88c6ece27fdb

@patchthecode

**Comment by antbald** [comment]
It is quite hard to follow you.. the library  is currently used in most of my projects and right now all my customers are going nuts (rightfully).
Can you please help me and describe all the steps that I need to follow to make the library functioning again?

**Comment by felipeflorencio** [comment]
The workaround way?

Go to JTACMonthViewProtocols, is the library protocol, as most probably you are using Pods, just go to the extension that has the method "didSelectDate" and comment the implementation from the extension, done your method will start being called again

Let me know if worked ;)

**Comment by felipeflorencio** [comment]
Or you can just replace in your pod file my fork, already have this specific workaround until the library owner adjust, just replace the pod for the library to this:
  pod 'JTAppleCalendar', :git => 'https://github.com/felipeflorencio/JTAppleCalendar.git', :tag => '1.1'
@antbald

**Comment by antbald** [comment]
Thx for helping me Felipe. So far the first suggestion you gave to me didn’t fix the problem. I have to try with the second one and hope. I will keep you updated. Thank you very much for helping me

**Comment by antbald** [comment]
Do you have any other ideas in case the last one won’t work?

**Comment by felipeflorencio** [comment]
depend of the problem, the click problem this fix, but if still happen the problem you need to check again your delegate and datasource if is properly setup as this last update a lot have being changed

**Comment by antbald** [comment]
Got it. I’m using the latest version of your library. Any important things about the setup (in the last update) I should be particularly careful or aware of?

**Comment by patchthecode** [comment]
ok ok.
I do not know what the problem is.

What version are you using?
What version of XCode are you using?
What version of Swift are you using?

**Comment by patchthecode** [comment]
The library cant just work one minute and fail the next. Can you let me know what changed?

**Comment by felipeflorencio** [comment]
The test beside my project this issue you can see just download / clone the last version
Xcode 10.2.1
Version: Last one
Swift: 4.2 or higher

**Comment by felipeflorencio** [comment]
@patchthecode if you want just add me to some channel and we can discuss. 

But actually looking what is the type of the delegate I actually expect this behaviour after looking for the implementation.

**Comment by antbald** [comment]
“The library cant just work one minute and fail the next. Can you let me know what changed?”

I know that can sounds crazy but is just what happened.
No update in the library and in the code of the app for 4 weeks and then, at 10:00 AM (Rome) the library just stop working. It is just crazy.
Let me gather all the details you need so can I update u

**Comment by patchthecode** [comment]
@felipeflorencio 

I do not understand what you are saying.
Are you also experiencing the issue that @antbald is mentioning? 

And just to be clear -- The issue is  --> Cells are unclickable??

**Comment by patchthecode** [comment]
@antbald 

just to confirm, cells are unclickable?

If so, do you have anything in "ShouldSelect" ?
let me know all details, because once a code is compiled and on device, it cant just stop working.

**Comment by felipeflorencio** [comment]
@antbald Found fix for now, it's the Swift version of your project, just need to change to 4.2

If you change in your project / target : Build Settings -> Swift Compiler Language -> Swift Language Version

Will start calling your methods and not the extension from the library

**Comment by ijameelkhan** [comment]
I updated to lastest code still my delegate functions are not called - please help

**Comment by patchthecode** [comment]
@ijameelkhan please tell me the version you are using.
1. Latest 7.1.7 ?
latest master?
2. Also you need to give more information about what XCode and swift version you are using?
3. What does not working mean?
Without knowing this, i cannot help.


If you are using Xcode11 Beta with swift 5, then there is a issue I have to fix which causes cells to not be tapped.

**Comment by ijameelkhan** [comment]
Latest 7.1.7
and Master , I tried pointing directly to master as well no luck.
Xcode 10.1

My Cells are just not clickable. It was working and suddenly it stoped working. 
Event the item itemDidSelect in the super class is not being called.

**Comment by ijameelkhan** [comment]
Like this function is not called either , which would explain my own methods not being called

<img width="1027" alt="Screenshot 2019-06-26 at 7 17 09 AM" src="https://user-images.githubusercontent.com/1882875/60146272-6f774000-97e2-11e9-840b-6d39d418c9bb.png">

**Comment by patchthecode** [comment]
@ijameelkhan do you have facebook as one of your pods?

**Comment by ijameelkhan** [comment]
Yes I do and I did suspect Facebook SDK when my code randomly jumped to some method swizzling inside there SDK.

**Comment by patchthecode** [comment]
yes, facebook is the cause as reported by other users. However, i heard someone tell me that the problem went away. Here is what he said:

> Guys, I’ve solved changing the Facebook v. from 4.35.0 to 4.39.1

Can you confirm?

**Comment by ijameelkhan** [comment]
Ok going to try it and report back.

**Comment by patchthecode** [comment]
I honestly dont know what facebook is doing. Isnt method swizzling sort of a hack? Doing this might mess up other libraries :/

**Comment by patchthecode** [comment]
Can you let me know what is the link to the FaceBook pod? Let me look myself.

**Comment by ijameelkhan** [comment]
Here is the proper fix: 
https://github.com/patchthecode/JTAppleCalendar/pull/1062
we shouldn't override stuff on super class.

**Comment by patchthecode** [comment]
I thought so too, but then look at his last comment.
He said it stopped working. I'll have to spend some time this weekend to take a deep look.

**Comment by patchthecode** [comment]
@ijameelkhan  By the way, the reason why i tried to those on the superclass is becase i did not want developers using it accidentally. I got many issues with people using them accidentally. I'll have to think on this.

**Comment by patchthecode** [comment]
@felipeflorencio 
found the issue.
it was my mistake.
The function was renamed.
If you are using master branch, please rename the function to 
```swift
func calendar(_ calendar: JTACMonthView, didSelectDate date: Date, cell: JTACDayCell?, cellState: CellState, indexPath: IndexPath)
```

An index path was added
